### PR TITLE
Add pad3d, add nabla spacing & tensorflow support for v2.x

### PR DESCRIPTION
**Pull request by khammernik**
This PR contains following updates:
- Add macro to `pytorch/CMakeLists.txt` to finish libs
- Add macro to `tensorflow/CMakeLists.txt` to finish libs
- Fix typo in `optoth/pad2d.py` pad2d_tranpose -> pad2d_transpose
- Add `pad3d` operator
- Add tensorflow wrapper for activations and padding operator
- Add spacing to nabla 2d/3d operator (anisotropic)

### CODE REVIEW

**Comment by khammernik** [comment]
Note: Framework builds for TF > 1.15, unittests work for TF 2.x.

File: python/optopy/nabla.py [hunk 1]
```diff
@@ -15,27 +15,27 @@
 # to run execute: python -m unittest [-v] optopy.nabla
 class TestNablaFunction(unittest.TestCase):
     
-    def _get_nabla_op(self, dtype, dim):
+    def _get_nabla_op(self, dtype, dim, hx, hy, hz):
```
Comment (koblererich) [inline python/optopy/nabla.py]: Great work! However, I do not like the interface that hz is necessary for a 2d nabla operatpr

**Review by koblererich — CHANGES_REQUESTED** [review]
Great work Kerstin! I like the macros and good that you fixed some bugs in the nabla op. 
However, it would be better if Nabla2d does not require an hx option.

File: python/optopy/nabla.py (hunk 1 above)
Comment (khammernik) [reply python/optopy/nabla.py]: Default arguments added. The 2d nabla op can be now called with `float_2d(hx, hy)` or `float_2d()`. The `_get_nabla_op` function is only used for unittesting.
